src/views/compute/KubernetesServiceTab.vue

```diff
@@ -327,14 +327,14 @@ export default {
           this.resource.projectid !== '') {
           params.projectid = this.resource.projectid
         }
-        if (this.isValidValueForKey(this.resource, 'associatednetworkid')) {
-          params.associatednetworkid = this.resource.associatednetworkid
+        if (this.isValidValueForKey(this.resource, 'networkid')) {
+          params.associatednetworkid = this.resource.networkid
         }
       }
       api('listPublicIpAddresses', params).then(json => {
         const ips = json.listpublicipaddressesresponse.publicipaddress
         if (this.arrayHasItems(ips)) {
-          this.publicIpAddress = ips[0]
+          this.publicIpAddress = ips.filter(x => x.issourcenat)[0]
```

Comment:
Does this break on shared networks, are shared networks supported for CKS @shwstppr? What about VPCs?

Reply:
@rhtyd we support shared network. VPC isn't supported at the moment

Reply:
@davidjumani as Rohit pointed this might break for shared networks as we don't need source NAT for k8s cluster. Were you able to test that?

Reply:
From what I've seen, it does use the source nat ip of the network
https://github.com/apache/cloudstack/blob/master/plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterStartWorker.java#L376